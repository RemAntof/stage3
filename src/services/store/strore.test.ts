import { configureStore } from '@reduxjs/toolkit';
import api from '@services/API/redux';
import itemsSelectedReducer from '@components/flyout/cardsSlice';
import { RootState } from './store';

describe('Redux Store', () => {
  it('should match the snapshot', () => {
    const storeInstance = configureStore({
      reducer: {
        [api.reducerPath]: api.reducer,
        itemsSelected: itemsSelectedReducer,
      },
      middleware: (getDefaultMiddleware) =>
        getDefaultMiddleware().concat(api.middleware),
    });
    const state: RootState = storeInstance.getState();
    expect(state).toMatchSnapshot();
  });
});
