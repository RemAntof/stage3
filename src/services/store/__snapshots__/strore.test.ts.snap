// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Redux Store should match the initial state snapshot 1`] = `
{
  "api": {
    "config": {
      "focused": true,
      "invalidationBehavior": "delayed",
      "keepUnusedDataFor": 60,
      "middlewareRegistered": false,
      "online": true,
      "reducerPath": "api",
      "refetchOnFocus": false,
      "refetchOnMountOrArgChange": false,
      "refetchOnReconnect": false,
    },
    "mutations": {},
    "provided": {},
    "queries": {},
    "subscriptions": {},
  },
  "itemsSelected": {
    "items": {
      "item": undefined,
    },
  },
}
`;

exports[`Redux Store should match the snapshot 1`] = `
{
  "api": {
    "config": {
      "focused": true,
      "invalidationBehavior": "delayed",
      "keepUnusedDataFor": 60,
      "middlewareRegistered": false,
      "online": true,
      "reducerPath": "api",
      "refetchOnFocus": false,
      "refetchOnMountOrArgChange": false,
      "refetchOnReconnect": false,
    },
    "mutations": {},
    "provided": {},
    "queries": {},
    "subscriptions": {},
  },
  "itemsSelected": {
    "items": {
      "item": undefined,
    },
  },
}
`;
